### PR TITLE
[FIX JENKINS-30785] Generalize some CLI stuff to AbstractItem

### PR DESCRIPTION
**Pull request by daniel-beck**
[JENKINS-30785](https://issues.jenkins-ci.org/browse/JENKINS-30785)

### CODE REVIEW

**Comment by daniel-beck** [comment]
@pjanouse PTAL

**Comment by daniel-beck** [comment]
*grumbles* stupid tests…

File: core/src/main/java/hudson/cli/ReloadJobCommand.java [hunk 1]
```diff
@@ -78,11 +79,10 @@ protected int run() throws Exception {
                 }
 
                 if(job == null) {
-                    // TODO: JENKINS-30785
-                    AbstractProject project = AbstractProject.findNearest(job_s);
+                    AbstractItem project = Items.findNearest(AbstractItem.class, job_s, jenkins);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/cli/ReloadJobCommand.java]: Why not `Item` then?

File: core/src/main/java/hudson/model/AbstractItem.java [hunk 2]
```diff
@@ -752,11 +752,11 @@ public String getSearchName() {
      */
     @CLIResolver
     public static AbstractItem resolveForCLI(
-            @Argument(required=true,metaVar="NAME",usage="Job name") String name) throws CmdLineException {
+            @Argument(required=true,metaVar="NAME",usage="Item name") String name) throws CmdLineException {
         // TODO can this (and its pseudo-override in AbstractProject) share code with GenericItemOptionHandler, used for explicit CLICommand’s rather than CLIMethod’s?
         AbstractItem item = Jenkins.getInstance().getItemByFullName(name, AbstractItem.class);
         if (item==null) {
-            AbstractProject project = AbstractProject.findNearest(name);
+            AbstractItem project = Items.findNearest(AbstractItem.class, name, Jenkins.getInstance());
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/AbstractItem.java]: Uh. IMHO should be reported as a separate bug

**Comment by jglick** [comment]
Will cover the `ReloadJobCommand` patch in #2874, where I can give it an integration test.

**Comment by jglick** [comment]
Actually nah, I will just merge yours in, looks good to me.

**Comment by jglick** [comment]
GitHub would have automatically marked this as merged if @oleg-nenashev had not squashed commits, losing history of what was actually included.

**Comment by oleg-nenashev** [comment]
@jglick Yes, I caused some mess in this case. squashing was not intentional this time since I didn't want to deal with mess in downstream PRs. But yeah, now it's my responsibility to deal with it

**Pull request closed without merging**